lms/static/js/student_account/views/RegisterView.js

```diff
@@ -65,6 +67,56 @@
                     this.listenTo(this.model, 'validation', this.renderLiveValidations);
                 },
 
+
+                buildForm: function(data) {
+                    var html = [],
+                        i,
+                        len = data.length,
+                        requiredFields = [],
+                        optionalFields = [],
+                        fieldTpl = this.fieldTpl;
+
+                    this.fields = data;
+
+                    for (i = 0; i < len; i++) {
+                        if (data[i].errorMessages) {
+                            // eslint-disable-next-line no-param-reassign
+                            data[i].errorMessages = this.escapeStrings(data[i].errorMessages);
+                        }
+
+                        if (data[i].required) {
+                            requiredFields.push(data[i]);
+                        } else {
+                            optionalFields.push(data[i]);
+                        }
+                    }
+
+                    html.push('<div class="required-fields">');
+                    for (i = 0; i < requiredFields.length; i++) {
+                        html.push(HtmlUtils.template(fieldTpl)($.extend(requiredFields[i], {
+                            form: this.formType,
+                            requiredStr: this.requiredStr,
+                            optionalStr: this.optionalStr,
+                            supplementalText: requiredFields[i].supplementalText || '',
+                            supplementalLink: requiredFields[i].supplementalLink || ''
+                        })));
+                    }
+                    html.push('</div>');
+                    html.push('<div class="optional-fields">');
+
+                    for (i = 0; i < optionalFields.length; i++) {
+                        html.push(HtmlUtils.template(fieldTpl)($.extend(optionalFields[i], {
+                            form: this.formType,
+                            requiredStr: this.requiredStr,
+                            optionalStr: this.optionalStr,
+                            supplementalText: optionalFields[i].supplementalText || '',
+                            supplementalLink: optionalFields[i].supplementalLink || ''
+                        })));
+                    }
+                    html.push('</div>');
```

Comment:
Nitpick: the HTML generated in lines 105-116 are almost the same as lines 94-104 so it could be good to move to a ```generateFields``` function that takes a ```className``` and ```fields``` array.